// Person message for udaconnect app
// Compile with:
// python3 -m grpc_tools.protoc -I./ --python_out=./ --grpc_python_out=./ person.proto
syntax = "proto3";

message Person {

  int32 id = 1;
  string first_name = 2;
  string last_name = 3;
  string company_name = 4;

}

message PersonList {

  repeated Person persons = 1;

}

message PersonRequest {

  int32 id = 1;

} 

message PersonListRequest {}

service PersonService {
  rpc GetAll(PersonListRequest) returns (PersonList);
  rpc Get(PersonRequest) returns (Person);
}

